### PR TITLE
Fix 309 incorrect number of systems when loading game

### LINKED ISSUE #309: Game loads with incorrect number of systems
### How to reproduce
* Start a new game with 4 number of systems
* Save and exit the game
* Start the game again, and continue
* Open the map: there will be 2 systems (the default) instead of 4

It also seems the game creates two lastShip.ini files inside the development environment, one inside gradle's out directory, and one inside the engine/src/main folder.

### Log details and game version
Newest develop commit c115c146a7378756157f70a4fea9d9aa437baafd

### PR DESCRIPTION
**Pull request by Avalancs**
# Description
Fixes #309 
* introduced a WorldConfig class to be able to remove the static SYS_COUNT variable which was not updated when loading a save game.
* changed world file to be json and include the number of systems
* moved around constants and functions responsible for saving to uphold encapsulation and single responsibility principle
* tutorial is now started as 'new game' to be future proof

# Testing
1. Start a new game with 2 systems; open the map and memorize layout
2. Start a new game with 2 systems; open the map and verify the layout of the systems are different
3. Start a new game with 4 systems; open the map and memorize layout
4. Go back to the menu
5. Click on 'Continue' and verify there are 4 systems and the layout is the same
5. Exit and Restart the game; continue; verify there are 4 systems and the layout is the same
6. Start a new game with 3 planets; verify there are only 3 planets
7. Start a tutorial. The number of planets will probably be 3, but it does not matter. Memorize the layout
8. Start another tutorial. Verify that the layout is different

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/SolApplication.java [hunk 1]
```diff
@@ -20,22 +20,19 @@
...
 import org.destinationsol.game.context.internal.ContextImpl;
-import org.destinationsol.assets.audio.OggMusicManager;
-import org.destinationsol.assets.audio.OggSoundManager;
 import org.destinationsol.menu.MenuScreens;
-import org.destinationsol.ui.DebugCollector;
-import org.destinationsol.ui.FontSize;
-import org.destinationsol.ui.SolInputManager;
-import org.destinationsol.ui.SolLayouts;
-import org.destinationsol.ui.UiDrawer;
+import org.destinationsol.ui.*;
```
Comment (Cervator) [inline engine/src/main/java/org/destinationsol/SolApplication.java]: Please avoid wildcard imports :-) Can be configured in IntelliJ to not collapse them

File: engine/src/main/java/org/destinationsol/game/SaveManager.java [hunk 2]
```diff
@@ -20,32 +20,28 @@
...
+import com.google.gson.stream.JsonReader;
 import org.destinationsol.IniReader;
+import org.destinationsol.common.SolRandom;
 import org.destinationsol.files.HullConfigManager;
-import org.destinationsol.game.item.Gun;
-import org.destinationsol.game.item.ItemContainer;
-import org.destinationsol.game.item.ItemManager;
-import org.destinationsol.game.item.MercItem;
-import org.destinationsol.game.item.SolItem;
+import org.destinationsol.game.item.*;
```
Comment (Cervator) [inline engine/src/main/java/org/destinationsol/game/SaveManager.java]: Couple more wildcard imports in here

File: engine/src/main/java/org/destinationsol/game/SaveManager.java [hunk 3]
```diff
@@ -191,4 +190,70 @@ public static ShipConfig readShip(HullConfigManager hullConfigs, ItemManager ite
...
+                writer.close();
+            }
+        }
+    }
+
+    /**
+     * Load the last saved world from file, or returns null if there is no file
+     */
+    public static WorldConfig loadWorld() {
+        if(SaveManager.resourceExists(WORLD_SAVE_FILE_NAME)) {
```
Comment (Cervator) [inline engine/src/main/java/org/destinationsol/game/SaveManager.java]: Trivial whitespace violation (space between `if` and opening `(` - might want to make sure you've got Checkstyle set up 👍

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Avalancs** [comment]
Set up checkstyle and disabled wildcards. Do you see any more outstanding issues with the formatting?

**Pull request merged**